client/allocwatcher/alloc_watcher.go

```diff
@@ -587,6 +587,12 @@ func (p *remotePrevAlloc) streamAllocDir(ctx context.Context, resp io.ReadCloser
 				p.prevAllocID, p.allocID, err)
 		}
 
+		if escapes, err := escapingfs.PathEscapesAllocDir(dest, "", hdr.Name); err != nil {
+			return fmt.Errorf("error evaluating directory: %w", err)
+		} else if escapes {
+			return fmt.Errorf("archive contains directory that escapes alloc dir")
+		}
```

Comment:
The error message we're returning here says "directory", but at the point we're checking this it could be any kind of archive object, right?

Reply:
That's true nice catch, I edited the error message to specify `object` instead of `directory`